docs/designs/serving_design.md

```diff
@@ -0,0 +1,101 @@
+    return input_layers
+
+def custom_model(feature_columns):
+    input_layers = get_input_layer()
+    dense_feature = tf.keras.layers.DenseFeatures(feature_columns=feat_cols)(input_layers)
+    dense = tf.keras.layers.Dense(10, activation='relu')(dense_feature)
+    dense = tf.keras.layers.Dense(1, activation='sigmoid')(dense)
+    return tf.keras.models.Model(inputs=input_layers, outputs=dense)
+```
+
+Although all feature columns in TensorFlow can be used in ElasticDL, tf.feature_column.embedding_column is not recommended in ElasticDL. Because the embedding_column has a variable containing a large embedding table. In eager execution the model must get all the embedding parameters to train. It will bring a large inter-process communication overhead.
+
+## Export the model with elasticdl.layers.Embedding to SavedModel
+
+Using native Keras layers to define a model is more user-friendly than using custom layers in ElasticDL. However, it is inefficient to train a model with tf.keras.layers.Embedding. When the model executes the forward-pass computation for each mini-batch, it must get all embedding parameters from the parameter server (PS) even if the mini-batch only contains several embedding ids. So, the elastic.layers.Embedding is designed to improve the training efficiency in ElasticDL. Considering user-friendliness and training efficiency, we need to define a model with tf.keras.layers.Embedding and train the model with elasticdl.layers.Embedding. For the Sequential model and the Functional models, we can use tf.keras.models.clone_model to replace the tf.keras.layers.Embedding with elasticdl.layers.Embedding before training starts. For subclass model, we can replace the attribute of tf.keras.layers.Embedding type with elastic.layers.Embedding.
+
+```python
+def replace_keras_embedding_with_edl_embedding(model)
+    # replace the embedding layer for Sequential and functional API models
+    if isinstance(model, tf.keras.Sequential) or model._is_graph_network:
+        def _clone_function(layer):
+            if type(layer) == keras.layers.Embedding:
+                edl_layer = elastic.layers.Embedding(layer.output_dim)
+                return edl_layer
+            return layer
+        return keras.models.clone_model(model, clone_function=_clone_function)
+    else:
+    # replace embedding attribute for subclass model
+        for attr_name, attr_value in model.__dict__.items():
+            if type(attr_value) == keras.layers.Embedding:
+                setattr(model, attr_name, edl_Embedding(attr_value.output_dim))
+        return model
+```
+
+However, tf.saved_model.save can not export the replaced model to SavedModel. Because ElasticDL.Embedding use tf.py_function to invoke Rpc to interact with the parameter server. It is not mapped to any native TensorFlow op. As a result we choose to save the origin model with native keras embedding layer, replace the embedding parameters with the trained parameters of elasticdl.layers.embedding.
```

Comment:
can not -> cannot

Reply:
It seems that *cannot* is more popular. Updated. :)